app/controllers/users_controller.rb

```diff
@@ -0,0 +1,43 @@
+class UsersController < ApplicationController
+  before_action :require_signin, except: %i[new create]
+  before_action :require_correct_user
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+
+    if @user.save
+      session[:user_id] = @user.id
+      redirect_to @user, notice: 'Thanks for signing up!'
+    else
+      render :new
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
+    @past_events = @user.attended_events.where('event_date < ?', Time.now).order('event_date DESC')
+    @upcoming_events = @user.attended_events.where('event_date > ?', Time.now).order('event_date DESC')
```

Comment:
User scopes created in your event model here.

Reply:
There are the scopes for user's attended_events (Past, Upcoming) which you can see on User show page. 


![Screenshot 2020-10-23 at 8 52 20 PM](https://user-images.githubusercontent.com/47393794/97022406-b2c9f600-1571-11eb-8487-26a7878c528c.png)